pkg/csi/service/vanilla/controller.go

```diff
@@ -115,53 +106,152 @@ func (c *controller) Init(config *cnsconfig.Config, version string) error {
+				vc.Client.ServiceContent.About.ApiVersion, err)
+			return err
 		}
-		for _, targetFSDatastore := range c.manager.VcenterConfig.TargetvSANFileShareDatastoreURLs {
-			isFSEnabled := dsToFileServiceEnabledMap[targetFSDatastore]
-			if !isFSEnabled {
-				return logger.LogNewErrorf(log, "file service is not enabled on datastore %s specified in "+
-					"TargetvSANFileShareDatastoreURLs", targetFSDatastore)
+		// Check if vSAN FS is enabled for TargetvSANFileShareDatastoreURLs only if
+		// CSIAuthCheck FSS is not enabled.
+		if !isAuthCheckFSSEnabled && len(c.manager.VcenterConfig.TargetvSANFileShareDatastoreURLs) > 0 {
+			datacenters, err := vc.ListDatacenters(ctx)
+			if err != nil {
+				return logger.LogNewErrorf(log, "failed to find datacenters from VC: %q, Error: %+v", vc.Config.Host, err)
+			}
+			// Check if file service is enabled on datastore present in
+			// targetvSANFileShareDatastoreURLs.
+			dsToFileServiceEnabledMap, err := common.IsFileServiceEnabled(ctx,
+				c.manager.VcenterConfig.TargetvSANFileShareDatastoreURLs, vc, datacenters)
+			if err != nil {
+				return logger.LogNewErrorf(log, "file service enablement check failed for datastore specified in "+
+					"TargetvSANFileShareDatastoreURLs. err=%v", err)
+			}
+			for _, targetFSDatastore := range c.manager.VcenterConfig.TargetvSANFileShareDatastoreURLs {
+				isFSEnabled := dsToFileServiceEnabledMap[targetFSDatastore]
+				if !isFSEnabled {
+					return logger.LogNewErrorf(log, "file service is not enabled on datastore %s specified in "+
+						"TargetvSANFileShareDatastoreURLs", targetFSDatastore)
+				}
+			}
+		}
+		if isAuthCheckFSSEnabled {
+			log.Info("CSIAuthCheck feature is enabled, loading AuthorizationService")
+			authMgr, err := common.GetAuthorizationService(ctx, vc)
+			if err != nil {
+				log.Errorf("failed to initialize authMgr. err=%v", err)
+				return err
+			}
+			c.authMgr = authMgr
+			go common.ComputeDatastoreMapForBlockVolumes(authMgr.(*common.AuthManager),
+				config.Global.CSIAuthCheckIntervalInMin)
+			isvSANFileServicesSupported, err := c.manager.VcenterManager.IsvSANFileServicesSupported(ctx,
+				c.manager.VcenterConfig.Host)
+			if err != nil {
+				log.Errorf("failed to verify if vSAN file services is supported or not. Error:%+v", err)
+				return err
+			}
+			if isvSANFileServicesSupported {
+				go common.ComputeFSEnabledClustersToDsMap(authMgr.(*common.AuthManager),
+					config.Global.CSIAuthCheckIntervalInMin)
+			}
+		}
+	} else {
+		// Multi vCenter feature enabled
+		c.managers = &common.Managers{
+			CnsConfig:      config,
+			VcenterManager: vcManager,
+		}
+		c.managers.VcenterConfigs = make(map[string]*cnsvsphere.VirtualCenterConfig)
+		c.managers.VolumeManagers = make(map[string]cnsvolume.Manager)
+		// Get VirtualCenterManager instance and validate version.
+		vcenterconfigs, err := cnsvsphere.GetVirtualCenterConfigs(ctx, config)
+		if err != nil {
+			log.Errorf("failed to get VirtualCenterConfigs. err=%v", err)
```

Comment:
Can you use `logger.LogNewErrorf` in the new code?

Reply:
done.